test/fakeintake/server/server_test.go

```diff
@@ -42,6 +42,51 @@ func TestServer(t *testing.T) {
 		assert.Equal(t, "server not running", err.Error())
 	})
 
+	for _, tt := range []struct {
+		name         string
+		opt          Option
+		possibleUrl  []string
+		expectedAddr string
+	}{
+		{
+			name: "Make sure WithPort sets the port correctly",
+			opt:  WithPort(1234),
+			possibleUrl: []string{
+				"http://[::]:1234",
+				"http://0.0.0.0:1234",
+			},
+			expectedAddr: ":1234",
+		},
+		{
+			name: "Make sure WithAddress sets the port correctly",
+			opt:  WithAddress("127.0.0.1:3456"),
+			possibleUrl: []string{
+				"http://127.0.0.1:3456",
+			},
+			expectedAddr: "127.0.0.1:3456",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			fi := NewServer(tt.opt)
+			assert.Equal(t, tt.expectedAddr, fi.server.Addr)
+			fi.Start()
+			assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+				assert.True(collect, fi.IsRunning())
+				// Assert that the address is in possible addr
+				assert.Contains(collect, tt.possibleUrl, fi.URL())
+				resp, err := http.Get(fi.URL() + "/fakeintake/health")
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Here you are not testing that requests to the expected URL are replied to. You are testing that the address decided by the fakeintake works fine. 

No need to test that `fi.URL()` contains the expected address, rather send an http request to the expected address and assert that the fakeintake replies to it

Reply:
I think we can't send a get request to `:1234`. We need to add `127.0.0.1`.

Reply:
It's fine to use `127.0.0.1:1234`. Indeed your expected address is `127.0.0.1:1234`, because it listens on local host by default.

Reply:
I advise against using the URL provided by the server. It breaks the test boundary.

Reply:
As discussed in private, I replaced it by `localhost`

Reply:
As discussed a second time, we now default the address to `0.0.0.0`.